pkg/utils/stack_functions.go

```diff
@@ -0,0 +1,484 @@
+	deploymentsClient := clSet.AppsV1().Deployments(ns)
+	list, err := deploymentsClient.List(metav1.ListOptions{})
+	if err != nil {
+		return nil, err
+	}
+	// if err != nil {
+	// 	panic(err)
+	// }
+	for _, d := range list.Items {
+		app := models.DescribeStackAppsItems0{}
+
+		deployName := d.ObjectMeta.Labels["stack.appsody.dev/id"]
+		deployVersion := d.ObjectMeta.Labels["stack.appsody.dev/version"]
+
+		if deployName == name && deployVersion == version {
+			fmt.Println("matched")
+			fmt.Println("deployName:")
+			fmt.Println(deployName)
+			fmt.Println("deployVersions:")
+			fmt.Println(deployVersion)
+			fmt.Println("labels")
+			fmt.Println(d.ObjectMeta.Labels["stack.appsody.dev/id"])
+			fmt.Println(d.ObjectMeta.Labels["stack.appsody.dev/version"])
+
+			app.AppKubernetesIoInstance = d.ObjectMeta.Labels["app.kubernetes.io/instance"]
+			app.AppKubernetesIoManagedBy = d.ObjectMeta.Labels["app.kubernetes.io/managed-by"]
+			app.AppKubernetesIoName = d.ObjectMeta.Labels["app.kubernetes.io/name"]
+			app.AppKubernetesIoPartOf = d.ObjectMeta.Labels["app.kubernetes.io/part-of"]
+			app.AppKubernetesIoVersion = d.ObjectMeta.Labels["app.kubernetes.io/version"]
+
+			apps = append(apps, &app)
+		}
+	}
+	return apps, err
+}
+
+func getApps(ns string, name string, version string) ([]*models.DescribeStackAppsItems0, error) {
+	fmt.Println("in getApps():")
+	fmt.Println("name:")
+	fmt.Println(name)
+	fmt.Println("version:")
+	fmt.Println(version)
+
+	apps := []*models.DescribeStackAppsItems0{}
+
+	var targetnamespaceList []string
+	k, err := getKabanero()
+	if err != nil {
+		return nil, err
+	}
+	// if err != nil {
+	// 	panic(err)
+	// }
+
+	targetnamespaceList = k.Spec.TargetNamespaces
+
+	myapps, err := getDeployments(name, version, ns)
+
+	apps = append(apps, myapps[0])
+
+	for _, nspace := range targetnamespaceList {
+		deployapps, err := getDeployments(name, version, nspace)
+		if err != nil {
+			return nil, err
+		}
+		// if err != nil {
+		// 	panic(err)
+		// }
+		for _, application := range deployapps {
+			apps = append(apps, application)
+		}
+	}
+	return apps, err
+}
+
+// list all stacks in namespace
+func ListStacksFunc() ([]*models.KabaneroStack, error) {
+	fmt.Println("Entered ListStacksFunc!")
+
+	ns := getHookNamespace()
+
+	stacksUnstructured, err := getStacks()
+
+	listOfStacks := []*models.KabaneroStack{}
+	for _, onestack := range stacksUnstructured.Items {
+		fmt.Println("Stack:")
+		stack := models.KabaneroStack{}
+		oneStackBytes, err := onestack.MarshalJSON()
+		if err != nil {
+			panic(err.Error())
+		}
+		var kabstack kabanerov1alpha2.Stack
+		json.Unmarshal(oneStackBytes, &kabstack)
+		stack.Name = kabstack.GetName()
+		fmt.Println("Stack Name:")
+		fmt.Println(stack.Name)
+		items := []*models.KabaneroStackStatusItems0{}
+		for _, dStackStatusVersion := range kabstack.Status.Versions {
+			item := models.KabaneroStackStatusItems0{}
+			fmt.Println(dStackStatusVersion.Status)
```

Comment:
I'm guessing you're going to come thru here and clean up these prints eventually.... the log is going to be pretty big.

Reply:
In the previous service, Alan wanted the prints to stay, to have some kind of logging.  Also it wraps after awhile I believe in the pod